server/routes/task.js

```diff
@@ -24,33 +32,98 @@ router.post("/create/" , async(req , res) => {
     return res.status(200).send("Task Added");
 });
 
-router.post("/picktask/" , async(req , res) => {
+router.post("/pick" , async(req , res) => {
     let task = await Task.findById(req.body.task.toString()).clone();
+    
     if(!task)
-        return res.status(500).send("No such task found");
-    if(task.completionStatus != "not started")
-        return res.status(500).send("task already picked by some team member");
-    let curTime = Date.now();
+        return res.status(500).send("No such task found!");
+    
+        if(task.completionStatus != "Available")
+        return res.status(500).send("Task already picked by some team member!");
+    
+        let curTime = Date.now();
+
     task.executedBy = req.body.member.toString();
-    task.completionStatus = "in progress";
+    task.completionStatus = "Ongoing";
     task.startTime = curTime;
     await task.save();
     return res.status(200).send("Task successfully picked!");
 });
 
-router.post("/finishtask/" , async(req , res) => {
+router.post("/submit" , async(req , res) => {
+
     let task = await Task.findById(req.body.task.toString()).clone();
+    
     if(!task)
-        return res.status(500).send("No such task found");
-    if(task.completionStatus != "in progress")
-        return res.status(500).send("task not in progress");
+        return res.status(404).send("No such task found!");
+        
     if(task.executedBy != req.body.member.toString())
-        return res.status(500).send("Task not assigned to this user");
+        return res.status(409).send("Task assigned to different user!");
+    
+    if (task.completionStatus == "In Review")
+        return res.status(409).send("Task already submitted for review!")
+
+    if(task.completionStatus == "")
+        return res.status(409).send("Task not yet picked up!");
+ 
+    if(task.completionStatus == "Completed")
+        return res.status(409).send("Task already completed!");   
+
     let curTime = Date.now();
-    task.endTime = curTime;
-    task.completionStatus = "finished";
-    await task.save();
-    return res.status(200).send("Task completed");
+    
+    await Task.findByIdAndUpdate(
+
+        req.body.task,
+        
+        { endTime : curTime, completionStatus : "In Review" },
+        
+        async function (err, task)
+        {
+            if (err)
+                console.log(err)
+            else
+            {
+                console.log(task.team)
+                await Team.findByIdAndUpdate(
+                    task.team,
+                    { $push: {review: req.body.task} }
+                ).clone()
+            }
+        } 
+    ).clone();
+    
+    return res.status(200).send("Task sent for review!");
+});
+
+router.post("/accept", async(req, res) => {
+
+    let task = await Task.findById(req.body.task).clone()
+
+    if (!task)
```

Comment:
minor: Null check can be done use  isNull function from the Underscore library for a more reliable checking.

example:
const myVar = null;
console.log(_.isNull(myVar)); // true

Reply:
merging for now, will resolve in a later commit